src/swap/saga.test.ts

```diff
@@ -596,257 +596,168 @@ describe(swapSubmitPreparedSaga, () => {
-    expect(analyticsProps.gasFeeUsd).toBeCloseTo(
-      analyticsProps.approveTxGasFeeUsd + analyticsProps.swapTxGasFeeUsd,
-      8
-    )
-  })
+      swapPrepared: mockSwapPrepared,
+    },
+    {
+      network: Network.Ethereum,
+      networkId: NetworkId['ethereum-sepolia'],
+      sellTokenId: mockEthTokenId,
+      buyTokenId: mockUSDCTokenId,
+      sellTokenAddress: '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee',
+      buyTokenAddress: mockUSDCAddress,
+      feeCurrencySymbol: 'ETH',
+      swapPrepared: mockSwapPreparedEthereum,
+    },
+  ]
 
-  it('should complete swap on Ethereum', async () => {
-    jest
-      .spyOn(Date, 'now')
-      .mockReturnValueOnce(mockQuoteReceivedTimestamp + 2_500) // swap submitted timestamp
-      .mockReturnValue(mockQuoteReceivedTimestamp + 10_000) // before send swap timestamp
+  it.each(testCases)(
+    'should complete swap on $network',
+    async ({
+      network,
+      networkId,
+      sellTokenId,
+      buyTokenId,
+      sellTokenAddress,
+      buyTokenAddress,
+      feeCurrencySymbol,
+      swapPrepared,
+    }) => {
+      jest
+        .spyOn(Date, 'now')
+        .mockReturnValueOnce(mockQuoteReceivedTimestamp + 2_500) // swap submitted timestamp
+        .mockReturnValue(mockQuoteReceivedTimestamp + 10_000) // before send swap timestamp
+
+      await expectSaga(swapSubmitPreparedSaga, swapPrepared)
+        .withState(store.getState())
+        .provide(createDefaultProviders(network))
+        .not.put(swapApprove())
+        .put(swapExecute())
+        .put(
+          addStandbyTransaction({
+            context: {
+              id: 'id-swap/saga-Swap/Execute',
+              tag: 'swap/saga',
+              description: 'Swap/Execute',
+            },
+            __typename: 'TokenExchangeV3',
+            networkId,
+            type: TokenTransactionTypeV2.SwapTransaction,
+            inAmount: {
+              value: BigNumber('0.0102'), // guaranteedPrice * sellAmount
+              tokenId: sellTokenId,
+            },
+            outAmount: {
+              value: BigNumber('0.01'),
+              tokenId: buyTokenId,
+            },
+            transactionHash: mockSwapTxReceipt.transactionHash,
+          })
+        )
+        .put(
+          transactionConfirmed('id-swap/saga-Swap/Execute', {
+            transactionHash: mockSwapTxReceipt.transactionHash,
+            block: mockSwapTxReceipt.blockNumber.toString(),
+            status: true,
+          })
+        )
+        .call([publicClient[network], 'waitForTransactionReceipt'], { hash: '0x2' })
+        .run()
+
+      expect(mockViemWallet.signTransaction).toHaveBeenCalledTimes(2)
+      expect(mockViemWallet.sendRawTransaction).toHaveBeenCalledTimes(2)
+      expect(loggerErrorSpy).not.toHaveBeenCalled()
+
+      expect(ValoraAnalytics.track).toHaveBeenCalledTimes(1)
+      expect(ValoraAnalytics.track).toHaveBeenCalledWith(SwapEvents.swap_execute_success, {
+        toToken: sellTokenAddress,
```

Comment:
are these correct? i thought the sellToken was the fromToken?

Reply:
Good catch, I'll let Copilot refactor this and I'm guilty of not checking this important detail 🤦